staging/src/k8s.io/client-go/tools/reference/ref_test.go

```diff
@@ -0,0 +1,56 @@
+package reference
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type TestRuntimeObj struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+}
+
+func (o *TestRuntimeObj) DeepCopyObject() runtime.Object {
+	panic("die")
+}
+
+func TestGetReferenceRefVersion(t *testing.T) {
+	tests := []struct {
+		name               string
+		input              *TestRuntimeObj
+		expectedRefVersion string
+	}{
+		{
+			name: "api from selflink",
+			input: &TestRuntimeObj{
+				ObjectMeta: metav1.ObjectMeta{SelfLink: "/api/v1/namespaces"},
+			},
+			expectedRefVersion: "v1",
+		},
+		{
+			name: "foo.group/v3 from selflink",
+			input: &TestRuntimeObj{
+				ObjectMeta: metav1.ObjectMeta{SelfLink: "/apis/foo.group/v3/namespaces"},
+			},
+			expectedRefVersion: "foo.group/v3",
+		},
+	}
+
+	scheme := runtime.NewScheme()
+	scheme.AddKnownTypes(schema.GroupVersion{Group: "this", Version: "is ignored"}, &TestRuntimeObj{})
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ref, err := GetReference(scheme, test.input)
+			if err != nil {
+				t.Fatal(err)
```

Comment:
I think this should be 
```golang
t.Error(err)
return
```
otherwise if the first subtest fails, it won't run the other one

t.Fatal calls runtime.Goexit()

Reply:
> I think this should be
> 
> t.Error(err)
> return
> otherwise if the first subtest fails, it won't run the other one
> 
> t.Fatal calls runtime.Goexit()

I don't know how it works, but I just tried it.  Calling fatal only fatals the subtest